R/cal_min_max_date.R

```diff
@@ -0,0 +1,127 @@
+#' Calculate minimum and maximum date and time in the dataframe
+#'
+#' @description This function derives the earliest/latest ISO8601 datetime
+#'
+#' @param raw_dataset Raw source data frame
+#' @param date_variable Single character string. Name of the date variable
+#' @param time_variable Single character string. Name of the time variable
+#' @param val_type Single character string determining whether to look
+#'   for the earliest or the latest datetime combination. Permitted values:
+#'   "min", "max". Default to "min".
+#' @param date_format Format of source date variable
+#' @param time_format Format of source time variable
+#'
+#' @return Data frame with 2 columns: unique patient_number and datetime variable
+#'   column storing the earliest/latest datetime.
+#'
+#' @export
+#' @examples
+#' EX <- tibble::tribble(
+#'   ~patient_number,    ~EX_ST_DT, ~EX_ST_TM,
+#'   "001",           "26-04-2022",   "10:20",
+#'   "001",           "25-04-2022",   "10:15",
+#'   "001",           "25-04-2022",   "10:19",
+#'   "002",           "26-05-2022", "UNK:UNK",
+#'   "002",           "26-05-2022",   "05:59"
+#' )
+#'
+#' min <- cal_min_max_date(EX,
+#'   "EX_ST_DT",
+#'   "EX_ST_TM",
+#'   val_type = "min",
+#'   date_format = "dd-mmm-yyyy",
+#'   time_format = "H:M"
+#' )
+#'
+#' max <- cal_min_max_date(EX,
+#'   "EX_ST_DT",
+#'   "EX_ST_TM",
+#'   val_type = "max",
+#'   date_format = "dd-mmm-yyyy",
+#'   time_format = "H:M"
+#' )
+#'
+cal_min_max_date <- function(raw_dataset,
+                             date_variable,
+                             time_variable,
+                             val_type = "min",
+                             date_format,
+                             time_format) {
+  # Check if date is present in the raw data frame
```

Comment:
I think we need to check if `date_variable`, `time_variable` are a string of single character, in case users input a vector of variable names.